File: provisioning/scim/search-list-users/code-samples/searchViaGet2.js
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// OPTIONAL QUERY PARAMETERS
const queryParams = {
    //filter: '<ENTER VALUE>',
    //startIndex: 1,
    //count: 100
}

const SDK = require('ringcentral');
const rcsdk = new SDK({server: process.env.serverURL, appKey: process.env.clientId, appSecret: process.env.clientSecret});
const platform = rcsdk.platform();
platform.login({ username: process.env.username, extension: process.env.extension, password: process.env.password }).then(() => {
    platform.get(`/scim/v2/Users`, queryParams).then((r) => {
        // PROCESS RESPONSE
    });
});

File: provisioning/scim/search-list-users/code-samples/searchViaGet2.cs
// https://developers.ringcentral.com/my-account.html#/applications
// Find your credentials at the above url, set them as environment variables, or enter them below

// OPTIONAL QUERY PARAMETERS
SearchViaGet2Parameters searchViaGet2Parameters = new SearchViaGet2Parameters {
    //filter = "<ENTER VALUE>",
    //startIndex = 1,
    //count = 100
};

RestClient rc = new RestClient(
    Environment.GetEnvironmentVariable("clientId"),
    Environment.GetEnvironmentVariable("clientSecret"),
    false
);
await rc.Authorize(
    Environment.GetEnvironmentVariable("username"),
    Environment.GetEnvironmentVariable("extension"),
    Environment.GetEnvironmentVariable("password")
);
var r = await rc.Scim().Users().List(searchViaGet2Parameters);
// PROCESS RESPONSE
